lib/clients/ClientCache.js

```diff
@@ -0,0 +1,223 @@
+const AWS = require('aws-sdk');
+const joi = require('@hapi/joi');
+
+const BackbeatClient = require('./BackbeatClient');
+const { transportJoi } = require('../config/configItems.joi');
+
+const s3ProfileJoi = {
```

Comment:
I'm not sure it's such a good idea to spread this pattern of using joi as a type system. It seems to be designed for the input data use case. I didn't comment on the last PR but it's not something I'm comfortable with seeing more and more.

For this type of code where we control all entry points, it's easy to know that all parameters will be compliant. For example the host and port come from the configuration where they will have already been validated.

Some typescript or flow annotations would be very useful (and urgent IMO) later on, but we're not achieving a lot of safety with joi, as by the time errors are caught at runtime, it's already too late.